internal/cmd/gocdk/init.go

```diff
@@ -15,11 +15,101 @@
 package main
 
 import (
+	"bytes"
 	"context"
+	"flag"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
 
+	"gocloud.dev/internal/cmd/gocdk/internal/templates"
 	"golang.org/x/xerrors"
 )
 
 func init_(ctx context.Context, pctx *processContext, args []string) error {
-	return xerrors.New("not implemented")
+	f := newFlagSet(pctx, "init")
+	var modpath string
+	f.StringVar(&modpath, "module-path", "", "the module path for your project's go.mod file")
+	f.StringVar(&modpath, "m", "", "alias for --module-path")
+
+	if err := f.Parse(args); xerrors.Is(err, flag.ErrHelp) {
+		return nil
+	} else if err != nil {
+		return usagef("gocdk init: %w", err)
+	}
+
+	if f.NArg() != 1 {
+		return usagef("gocdk init [options] PATH")
```

Comment:
Can you clarify for me (and the user) what the difference between `PATH` here and `module-path` is? I'll probably figure it out as I read the code below but it's not clear from the help.

Reply:
`module-path` is the import path for the module we'll be creating, and `path` is the path to the directory we will be creating to put the project in. 

Do you think it would be clearer if `module-path` was named `import-path`?

Reply:
Where is it going to be used? In the `.mod` file, or ...?

Reply:
I get it now, it's the module path for the go.mod file. Maybe just update the help to indicate what the default is. Although not sure it will be possible to explain that succinctly.

Overall this is fine for now but I'm concerned that users will have a hard time understanding it.